### DISCUSSION TITLE
How to add a custom threshold map

### DISCUSSION THREAD

**Discussion by derekjohn37** [dlemstra/Magick.NET#1005] [Help] [answered]
ImageMagick has a resource configuration file for Threshold Maps for Ordered Posterized Dithering: "thresholds.xml". I would like to add a custom map for use with Magick.NET. Can this be done? (i.e. append to the thresholds.xml file) If so, any ideas how?

**Comment by dlemstra** [comment, marked as answer]
You can do that with the `ConfigurationFiles` class. Below is an example of how you can change the Threshold.Data configuration:

```C#
var configuration = ConfigurationFiles.Default;
configuration.Thresholds.Data = "<yourxml/>";

MagickNET.Initialize(configuration);
```

Make sure you call `MagickNET.Initialize` at the start of your program.

**Reply by derekjohn37** [reply]
Thanks @dlemstra that worked perfectly and steered me to the ConfigurationFiles class for future ref. Much appreciated!
